lib/proxy/archive_extract.rb

```diff
@@ -0,0 +1,30 @@
+module Proxy
+  class ArchiveExtract < Proxy::Util::CommandTask
+    include Util
+
+    def initialize(image_path, file_in_image, dst_path)
+      args = [which('isoinfo')]
+
+      # read the file
+      args << "-R"
+      # set image path
+      args += ["-i", image_path.to_s]
+      # set file path within the image
+      args += ["-x", file_in_image.to_s]
```

Comment:
Performance wise it's probably better to instantiate complete. You could do this:
```ruby
args = [
  which('isoinfo'),
  # Print information from Rock Ridge extensions
  '-R',
  # Filename to read ISO-9660 image from
  '-i', image_path.to_s,
  # Extract specified file to stdout
  '-x', file_in_image.to_s,
]
```

Reply:
Done.